dotnet/targets/Xamarin.Shared.Sdk.props

```diff
@@ -105,4 +126,22 @@
 	<PropertyGroup Condition="'$(MtouchUseLlvm)' == '' And '$(Configuration)' == 'Release' And ('$(_PlatformName)' == 'iOS' Or '$(_PlatformName)' == 'tvOS')">
 		<MtouchUseLlvm>true</MtouchUseLlvm>
 	</PropertyGroup>
+
+	<!-- Various options when using NativeAOT -->
+	<PropertyGroup Condition="'$(PublishAot)' == 'true' And '$(_IsPublishing)' == 'true'">
+		<!-- Disable our own assembly IL stripping logic, because ILC does that already -->
+		<EnableAssemblyILStripping>false</EnableAssemblyILStripping>
+
+		<!-- We're using our own native main function when using NativeAOT -->
+		<CustomNativeMain>true</CustomNativeMain>
+
+		<!-- We must find the BCL libraries using the runtime pack instead of using the built-in NativeAOT BCL -->
+		<PublishAotUsingRuntimePack>true</PublishAotUsingRuntimePack>
+
+		<!-- This turns off some NativeAOT logic we don't want nor need -->
+		<NativeCompilationDuringPublish>false</NativeCompilationDuringPublish>
+
+		<!-- This works around an issue in NativeAOT: https://github.com/dotnet/runtime/issues/86186 -->
+		<IlcKeepManagedDebuggerSupport>true</IlcKeepManagedDebuggerSupport>
```

Comment:
`IlcKeepManagedDebuggerSupport` should be fixed in latest SDK, and this should be unnecessary.

Reply:
I tried earlier today, and things were still broken, so I put the workaround back in for now.

Reply:
https://github.com/dotnet/runtime/commit/652433bfe9858f7f4d6f6ffcf0c994cb9b91e482 (for reference). There is some arcade blockage in dotnet/runtime (still not fully resolved) so I imagine it may not have flown through yet...

Reply:
Filed to look at it after this PR is in: https://github.com/xamarin/xamarin-macios/issues/18478